srcStatic/Log.h

```diff
@@ -7,4 +7,19 @@ class Logger;
 
 
 void SetLogger(Logger* logger);
+void SetLoggerError(Logger* newLogger);
+
 void Log(const std::string& message, const std::string& moduleName = "op2ext.dll");
+
+
+// Outputs a debug string. Output is not logged by the logger.
+void LogDebug(const std::string& message);
+
+// Logs an error message with the logger and then posts it to user in a modal dialog box.
+void PostErrorMessage(const std::string& errorMessage, const std::string& sourcePathFilename, long lineInSourceCode);
+
+
+// Allow defaulting of the __FILE__ and __LINE__ macros at the point of call
+// Macro name must be different from original function name
+// Using the same name will cause an error when the compiler sees the underlying function definition
+#define PostError(errorMessage) PostErrorMessage(errorMessage, __FILE__, __LINE__)
```

Comment:
I would be okay removing the idea of recording the file and code line number the error comes from. While it can be helpful, it isn't worth keeping around if it gets in the way of a quality log system.

Reply:
I would like to keep the information. Maybe there's a way we could keep it that avoids irregularities in the API. Right now it creates irregularities in the naming of the core methods, and the number and types of parameters they accept. I guess I would be open to removing it as well. Probably a topic for a future branch.